aggregator/src/aggregator.rs

```diff
@@ -1761,358 +1755,318 @@ impl VdafOps {
-                                // However, this is desirable: this can only happen if we receive
-                                // too many report shares in an aggregation job for us to store,
-                                // which is a whole-aggregation problem rather than a per-report
-                                // problem. (separately, this would require more than u64::MAX
-                                // report shares in a single aggregation job, which is practically
-                                // impossible.)
-                                ord.try_into()?,
-                                Some(PrepareResp::new(
-                                    *prepare_init.report_share().metadata().id(),
-                                    prepare_step_result,
-                                )),
-                                report_aggregation_state,
-                            ),
-                            output_share,
+                                prepare_step_result,
+                            )),
+                            report_aggregation_state,
                         ),
-                    });
+                        output_share,
+                    ),
                 }
-                Ok(report_share_data)
-            }
-        };
-        let (sender, receiver) = oneshot::channel();
-        rayon::spawn(|| {
-            // Do nothing if the result cannot be sent back. This would happen if the initiating
-            // future is cancelled.
-            //
-            // We need to catch panics here because rayon's default threadpool panic handler will
-            // abort, and it would be preferable to propagate the panic in the original future to
-            // avoid behavior changes.
-            //
-            // Using `AssertUnwindSafe` is OK here, because the only interior mutability we make use
-            // of is OTel instruments, which can't be put into inconsistent states merely by
-            // incrementing counters, and the global HPKE keypair cache, which we only read from.
-            // Using `AssertUnwindSafe` is easier than adding infectious `UnwindSafe` trait bounds
-            // to various VDAF associated types throughout the codebase.
-            let _ = sender.send(catch_unwind(AssertUnwindSafe(prep_closure)));
-        });
-        let report_share_data = receiver
-            .await
-            .map_err(|_recv_error: RecvError| {
-                Error::Internal("threadpool failed to send VDAF preparation result".into())
-            })?
-            .unwrap_or_else(|panic_cause: Box<dyn Any + Send>| {
-                resume_unwind(panic_cause);
-            })?;
-
-        // TODO: Use Arc::unwrap_or_clone() once the MSRV is at least 1.76.0.
-        let agg_param = Arc::try_unwrap(agg_param).unwrap_or_else(|arc| arc.as_ref().clone());
+            })
+            .collect_into_vec(&mut report_share_data);
```

Comment:
Dropping the `rayon::spawn()` and making this call from an async task results in regressing on poll times. I saw one poll between 8 and 17 seconds per aggregation job when testing with a large number of report shares.

Reply:
Fixed by merging the follow-on PR (lol).

Reply:
Looking much better, the worst-case poll time is now between a quarter second and a half second.